include/seastar/core/app-template.hh

```diff
@@ -153,6 +155,7 @@ public:
     boost::program_options::options_description& get_options_description();
     boost::program_options::options_description& get_conf_file_options_description();
     boost::program_options::options_description_easy_init add_options();
+    void set_app_visible_options_for_help(boost::program_options::options_description* opts);
```

Comment:
if we really want to accept `nullptr` as a valid input, probably should add a doxygen comment to explain what `nullptr` is for. and, mutating the value passed by caller and expect that caller to keep the referenced object alive might also warrant a comment.

still i think a reference is a better fit in this case.